### PR TITLE
Convert smartling integration to smartling tms

### PR DESCRIPTION
**Pull request by camertron**
With the new TMS abstraction comes the need to convert our existing Smartling integration. Apologies for the loooong PR :(

@jdoconnor @seunghyo @zvkemp

### CODE REVIEW

File: lib/rosette/tms/smartling-tms/configurator.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# encoding: UTF-8
+
+module Rosette
+  module Tms
+    module SmartlingTms
+
+      class Configurator
+        DEFAULT_PULL_EXPIRATION = 60 * 60  # one hour
```
Comment (11mdlow) [inline lib/rosette/tms/smartling-tms/configurator.rb]: Since you have the comment you can probably just have 3600 here

**Comment by 11mdlow** [comment]
Is most of this just moving/renaming files? There are a lot of deleted files and newly added ones...just wanted to make sure I'm looking at this right...

**Comment by camertron** [comment]
Yep, lots of moving and renaming. There's also Repository, Configurator, and TranslationMemoryDownloader that are new.

File: lib/rosette/tms/smartling-tms/translation_memory_downloader.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        def build_synchronously
+          repo_config.locales.each_with_object({}) do |locale, ret|
+            ret[locale.code] = download(locale)
+          end
+        end
+
+        def build_asynchronously
+          pool = Concurrent::FixedThreadPool.new(configurator.thread_pool_size)
+          hash_mutex ||= Mutex.new
```
Comment (11mdlow) [inline lib/rosette/tms/smartling-tms/translation_memory_downloader.rb]: is this an inherited `attr_accessor` from somewhere? why the memoization? is it just a concurrency thing?

**Comment by 11mdlow** [comment]
just a couple minor questions/comments :ship:

File: lib/rosette/tms/smartling-tms/translation_memory_downloader.rb (hunk 2 above)
Comment (camertron) [inline lib/rosette/tms/smartling-tms/translation_memory_downloader.rb]: Oops, no need for the `||=`, I'll take it out.

File: lib/rosette/tms/smartling-tms/configurator.rb (hunk 1 above)
Comment (camertron) [inline lib/rosette/tms/smartling-tms/configurator.rb]: Good call.

**Pull request merged**